tests/utils/test_output_utils.py

```diff
@@ -507,16 +509,26 @@ def test_precomputed_stats_to_uns(
     a_data.write_h5ad(h5ad_path)
 
     uns_key = 'serialization_test'
+    uns_key_nested = 'serialization_test_nested'
 
     precomputed_stats_to_uns(
         precomputed_stats_path=precomputed_stats_fixture,
         h5ad_path=h5ad_path,
         uns_key=uns_key)
 
-    roundtrip_path = uns_to_precomputed_stats(
-        uns_key=uns_key,
-        h5ad_path=h5ad_path,
-        tmp_dir=tmp_dir_fixture)
+    if not nested_uns:
+        roundtrip_path = uns_to_precomputed_stats(
+            uns_keys_list=[uns_key],
+            h5ad_path=h5ad_path,
+            tmp_dir=tmp_dir_fixture)
+    else:
+        a_data = anndata.read_h5ad(h5ad_path)
+        a_data.uns[uns_key] = {uns_key_nested: a_data.uns[uns_key]}
```

Comment:
Just being paranoid:

The way this code is constructed, the file at `h5ad_path` still has the expected data at `[uns_key]` which means that, even if `uns_to_precomputed_stats` were not modified, the test would still pass, I think.

Can you write a new anndata object at a different location that *only* has the expected data at your nested location. I am imagining something like this:

```
new_h5ad_path = mkstemp_clean(dir=tmp_dir_fixture, suffix='.h5ad')
src = anndata.read_h5ad(h5ad_path)
new_data = anndata.AnnData(
    obs=src.obs,
    var=src.var,
    X=src.X[()],
    uns = {uns_key_nested: {uns_key: {src.uns[uns_key]}}}
)
new_data.write_h5ad(new_h5ad_path)
h5ad_path = new_h5ad_path
```